include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -736,3 +736,23 @@ def RefSendOp: FIRRTLExprOp<"ref.send", [RefResultTypeConstraint<"base", "result
 
   let assemblyFormat = "$base attr-dict `:` qualified(type($base))";
 }
+
+def RefSubOp : FIRRTLExprOp<"ref.sub"> {
+  let summary = "Extract an element of an aggregate RefType value";
+  let description = [{
+    The refsub expression statically refers, by index, to a sub-element
+    of an expression with a RefType. The index must be a non-negative
+    integer and cannot be equal to or exceed the length of the underlying
+    bundle or vector size.
+    ```
+      %result = firrtl.refsub %input[index] : t1
+    ```
+    }];
+
+  let arguments = (ins RefType:$input, I32Attr:$index);
+  let results = (outs RefType:$result);
+
+  let assemblyFormat =
+     "$input `[` $index `]` attr-dict `:` qualified(type($input))";
+
+}
```

Comment:
The alternative design point is to use existing subfield/subaccess. I don't have a strong opinion here and am fine with `RefSubOp`.

Reply:
Yeah, it seemed like a good idea, to keep the `RefType` suported ops completely separate so that there is no ambiguity.